### ISSUE TITLE
[🐛BUG] DIN cannot accept more than one numerical feature

### ISSUE THREAD

**Issue by okandalan** [RUCAIBox/RecBole#1941]
Hello,

Thank you for your work. When I try to give more than one numerical feature to the DIN as input, I get this error:
"split_with_sizes expects split_sizes to sum exactly to 4 (input tensor's size at dimension -1), but got split_sizes=[1, 1]"

It is on this line in the forward method of the FLEmbedding class.
`base, index = torch.split(input_x, [1, 1], dim=-1)`

There is another issue (#1554 ) related to the same problem for SASRecF but it was closed. Thus, I opened a new one.

My yaml file:

```
#Environment Settings
seed: 2020
reproducibility: True
data_path: "dataset/"
dataset: RentTheRunway
show_progress: True
model: DIN

#Data settings
USER_ID_FIELD: user_id
ITEM_ID_FIELD: item_id
TIME_FIELD: timestamp

load_col:
  inter: [user_id, item_id, rating, timestamp, fit, rented_for]
  user: [user_id, age, bust_size, weight, height, body_type]
  item: [item_id, category, size]

MAX_ITEM_LIST_LENGTH: 50

numerical_features: [age, weight, size]

threshold:
  rating: 8

#Training Settings
epochs: 50
train_batch_size: 4096
learner: sgd
learning_rate: 3e-3
mlp_hidden_size: [256, 256, 256]

train_neg_sample_args:
  sample_num: none

#Evaluation Settings
eval_args:
  split: { "LS": "valid_and_test" }
  order: TO
  group_by: user
  mode: labeled

metrics: ["AUC", "LogLoss"]

eval_batch_size: 131072

```

**Comment by Yilu114** [comment]
According to your description, you encountered an error when trying to input more than one numerical feature into the DIN model in RecBole. The error occurred in the forward method of the FLEmbedding class. This error is due to the `torch.split` function expecting the sum of `split_sizes` to be exactly equal to the size of the input tensor at the specified dimension, but got `split_sizes=[1, 1]`.

In your yaml file, you specified `numerical_features: [age, weight, size]`, which means you are trying to input three numerical features. However, based on my understanding of RecBole, the DIN model may only support one numerical feature as input. This might be the cause of the error you encountered.

To solve this problem, you could try setting `numerical_features` to a list containing only one feature, such as `numerical_features: [age]`. Then, you can rerun your code to see if the same error occurs.

If you still encounter problems, or if you need to use multiple numerical features, I recommend you to refer to the [API documentation](https://recbole.io/docs/) of RecBole or open an issue in the GitHub repository of RecBole for more professional help.
